drivers/flash/spi_flash_at45.c

```diff
@@ -50,6 +61,8 @@ LOG_MODULE_REGISTER(spi_flash_at45, CONFIG_FLASH_LOG_LEVEL);
 
 #define INST_HAS_RESET_OR(inst) DT_INST_NODE_HAS_PROP(inst, reset_gpios) ||
 #define ANY_INST_HAS_RESET_GPIOS DT_INST_FOREACH_STATUS_OKAY(INST_HAS_RESET_OR) 0
+#define ANY_INST_WORKAROUND_READ (DT_ANY_INST_HAS_PROP_STATUS_OKAY(workaround_broken_read))
+#define ANY_INST_WORKAROUND_WRITE (DT_ANY_INST_HAS_PROP_STATUS_OKAY(workaround_broken_write))
```

Comment:
Specifically the read-modify-write is broken, the write works fine.

Reply:
Both the write (0x02) & modify (0x58) do not work on this particular chip and are address by this PR

```
/* - Main Memory Byte/Page Program through Buffer 1 without Built-In Erase */
#define CMD_WRITE		0x02
/* - Buffer 1 write */
#define CMD_WRITE_BUF1		0x84
/* - Buffer 1 to Main Memory Page Program without Built-In Erase */
#define CMD_BUF1_TO_MAIN	0x88
/* - Read-Modify-Write */
#define CMD_MODIFY		0x58
/* - Main Memory Page Program Through Buffer 1 */
#define CMD_MAIN_THRU_BUF1	0x82
/* - Main Memory Page to Buffer 1 Transfer */
#define CMD_MAIN_TO_BUF1	0x53
```

write: 0x02 will be replace by 0x84 + 0x88
modify: 0x58 will be replace by 0x82 + 0x53

___

Do you think that I should have two properties for that? i.e. `workaround-broken-write-02H` & `workaround-broken-modify-58H`

Reply:
I've modified the binding, but havent change the implementation, LMK what you think, thanks!

Reply:
I think this is OK. I hope we will not have to make more of these :D (at least not too many).